app/src/modules/settings/routes/data-model/fields/components/field-select.vue

```diff
@@ -205,6 +214,14 @@ export default defineComponent({
 
 		const nestedFields = computed(() => props.fields.filter((field) => field.meta?.group === props.field.field));
 
+		const relatedCollectionInfo = computed(() => getRelatedCollection(props.field.collection, props.field.field));
+
+		const showRelatedCollectionLink = computed(
+			() =>
+				relatedCollectionInfo.value !== null &&
+				['translations', 'm2o', 'm2m', 'o2m', 'files'].includes(localType.value as string)
+		);
```

Comment:
Let's hide the button for self-referencing relations.

```suggestion
		const showRelatedCollectionLink = computed(
			() =>
				relatedCollectionInfo.value !== null &&
				props.field.collection !== relatedCollectionInfo.value.relatedCollection &&
				['translations', 'm2o', 'm2m', 'o2m', 'files'].includes(localType.value as string)
		);
```

https://user-images.githubusercontent.com/26413686/234963252-559d8342-80e1-46b2-9f88-217ea86369bd.mov

Reply:
Good one